common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Note.scala

```diff
@@ -0,0 +1,26 @@
+package uk.ac.wellcome.models.work.internal
+
+import io.circe.{Encoder, Decoder}
+
+sealed trait NoteType
+
+case class Note(content: String, noteType: NoteType)
+
+object NoteType {
```

Comment:
[I think using the `withDescriminator` set here](https://github.com/wellcometrust/scala-json/blob/master/src/main/scala/uk/ac/wellcome/json/JsonUtil.scala#L15-L16), circe will do this automagically.

https://circe.github.io/circe/codecs/adt.html#the-future

Reply:
:thinking: to do it like that the code would be more like this:

```scala
sealed trait Note {
  val content: String
}

case class GeneralNote(val content: String) extends Note
```

which would serialise to:

```scala
case class GeneralNote(val content: String) extends Note

{
  "content": ".. note content ..",
  "type": "GeneralNote"
}
```

I agree much better to use the inbuilt serialisation rather than having to implement ourselves. I think I did it this way as was thinking of cases where we encounter a note type we don't recognise (the `Unknown` above), which would happen if we decided to remove a type but still had data in the index. It would error with the default behaviour, but maybe over thinking things and don't need to handle such cases.

Reply:
Yea, we shouldn't have types we don't recognise in the data - I would really like to avoid that situation, and having a hard error would handle that.

The data and model should always reflect each other, and if we have to remove a type, part of that would be thinking about what we do with the data.